test/application_full_test.rb

```diff
@@ -1,42 +1,104 @@
 require_relative "test_helper"
 
 class TestFullApplication < Minitest::Test
-  attr_reader :full_application, :env, :file, :controller
+  include Mocha::Integration::MiniTest
+  attr_reader :full_application, :env, :file, :controller,
+              :new_view, :show_view, :folder
 
   def setup
     @env = {
       "REQUEST_METHOD" => "GET",
-      "PATH_INFO" => "/users/new"
+      "PATH_INFO" => "/users/new",
+      "RACK_INPUT" => ""
     }
     set_up_full_app_and_routes
   end
 
   def set_up_full_app_and_routes
+    setup_instance_vars
+    prepare_route
+    create_controller_view_new
+    create_controller_view_show
+    create_controller_file
+  end
+
+  def setup_instance_vars
     @full_application = Hemp::Application.new
+    @controller = "users_controller.rb"
+    @folder = "app/views/users"
+    @new_view = "#{folder}/new.erb"
+    @show_view = "#{folder}/show.erb"
+  end
+
+  def prepare_route
     full_application.pot.prepare do
-      get "/users/new", to: "users#new"
+      resources :users
     end
-    @controller = "users_controller.rb"
-    create_controller_file controller
   end
 
-  def create_controller_file(controller)
+  def create_controller_file
     File.open(controller, "w") do |file|
-      file.write(
-        "class UsersController;"\
-          "def new;"\
-            "'new user created';"\
-          "end;"\
-        "end;"
-      )
+      file.write(controller_class_body)
       current_dir = File.expand_path("..", file)
       $LOAD_PATH.unshift current_dir
     end
   end
 
+  def create_controller_view_new
+    require 'fileutils'
+    FileUtils::mkdir_p folder
+    File.open(new_view, "w") do |file|
+      file.write(new_view_body)
+    end
+  end
+
+  def create_controller_view_show
+    require 'fileutils'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.